controllers/notifications/NotificationController.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file Controller RESTful Web service API for notifications resource
+ */
+
+import HttpRequest from '../shared/HttpRequest';
+import HttpResponse from '../shared/HttpResponse';
+import { Express, Router } from 'express';
+import { adaptRequest } from '../shared/adaptRequest';
+import { okResponse as okResponse } from '../shared/createResponse';
+import { isAuthenticated } from '../auth/isAuthenticated';
+import INotification from "../../models/notifications/INotification";
+import NotificationDao from '../../daos/notifications/NotificationsDao';
+
+
+/**
+ * Represents the implementation for handling the notifications resource api.
+ */
+export default class NotificationController {
+  private readonly notificationDao: NotificationDao;
+
+  /** Constructs the notifications controller with an notificationDao implementation. Defines the endpoint paths, middleware, method types, and handler methods associated with each endpoint.
+   *
+   * @param {NotificationDao} notification a notification dao used to find notifications resources in the database.
+   */
+  constructor(
+    path: string,
+    app: Express,
+    notificationDao: NotificationDao,
+  ) {
+    this.notificationDao = notificationDao;
+    const router = Router();
+    router.get(
+      '/notifications',
+      isAuthenticated,
+      adaptRequest(this.findAllNotifications)
+    );
+    router.get(
+      '/users/:userId/notifications',
+      isAuthenticated,
+      adaptRequest(this.findNotificationsForUser)
+    );
+    router.post(
+      '/users/:userId/notifications',
+      isAuthenticated,
+      adaptRequest(this.createNotificationForUser)
+    );
+    router.put(
+      '/notifications/:nid/read',
+      isAuthenticated,
+      adaptRequest(this.updateNotificationAsRead));
+    app.use(path, router);
+    Object.freeze(this); // Make this object immutable.
+  }
+
+  /**
+   * Processes the endpoint request of creating a notification by calling the notificationDao, 
+   * which will create and return a notification document. 
+   * @param {HttpRequest} req the request data containing client data
+   * @returns {HttpResponse} the response data to be sent to the client
+   */
+  createNotificationForUser = async (req: HttpRequest): Promise<HttpResponse> => {
+    const userNotifiedId = req.params.userId;
+    const type = req.body.type;
+    const userActing = req.body.userActing;
+
+    // new like
+    return {
+      body: await this.notificationDao.createNotificationForUser(type, userNotifiedId, userActing)
```

Comment:
I have a few of these in my code. We should probably consistently format like you have in the next method where you first define variable being returned from the DAO and then an okResponse(data)